pkg/config/remote/client.go

```diff
@@ -0,0 +1,201 @@
+package remote
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/api/security"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/data"
+	"github.com/DataDog/datadog-agent/pkg/config/remote/uptane"
+	"github.com/DataDog/datadog-agent/pkg/proto/pbgo"
+	agentgrpc "github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+// Client is a remote-configuration client to obtain configurations from the local API
+type Client struct {
+	sync.Mutex
```

Comment:
why promoting the Mutex if we anyway are going to use it only internally to the client?

Reply:
not really, it's just a pattern I picked up to write short `c.Lock()` instead of `c.mutex.Lock()`. I agree it's not great, especially with public structs